### ISSUE TITLE
Jump to definition inside installed libraries

### ISSUE THREAD

**Issue by ddickstein** [ocaml/dune#6397]
A while ago I discussed jumping to definition inside installed libraries, which would have required installing .merlin files alongside the libraries (https://github.com/ocaml/merlin/issues/863). Now that this is no longer a requirement since dune and Merlin can communicate directly without .merlin files, can this feature request be revisited?

**Comment by Alizter** [comment]
@voodoos Could you confirm what rgrinberg wrote on discuss will work:

> When merlin queries dune for the .merlin config, dune should be able to generation information that makes it possible to navigate through installed projects. I just need to confirm this is possible with Ulysse - the main developer of merlin/dune integration.

**Comment by voodoos** [comment]
What `dune ocaml-merlin` does right now is simply to check if a configuration was generated by Dune for the requested module and print that configuration to stdout. These configuration files are not installed so that doesn't work for installed modules.

Having `dune ocaml-merlin` fabricate a configuration in that case seems a sensible thing to do. However this will only be possible if enough information is installed along with the library files. Looking at the `dune-package` files that contain some information about the libraries, there is indeed some missing information. For example, the `flags` field of the libraries stanza is missing which means that merlin won't be able to correctly type some modules (missing a `-open ...` flag for example). I also expect preprocessing flags to be missing. These flags might be retrievable from the `cmt` files, but that sounds very hackish.

I think it might be relatively simple to use the information in the dune-package files to provide some sort of support for navigation in the installed world, by feeding Merlin the path to build artifacts. However, correct behavior for every library is probably not achievable with the currently installed artifacts.

**Comment by ddickstein** [comment]
What are the blockers to installing the artifacts necessary for correct
behavior in every installed library in a non-hacky way? If the concern is
disk space I’d certainly be willing to pay that cost - if it were gated by
a flag I’d happily pass it.

**Comment by Alizter** [comment]
Another solution is to perhaps take advantage of package management. When dune builds packages itself, it could make sure that merlin works everywhere as intended.

**Comment by voodoos** [comment]
> Another solution is to perhaps take advantage of package management. When dune builds packages itself, it could make sure that merlin works everywhere as intended.

Yes, Dune Pkg Management should solve that issue since every package will be built in the local project.
